shotover-proxy/src/frame/cassandra.rs

```diff
@@ -110,7 +115,7 @@ impl CassandraFrame {
                         params: body.query_params,
                     }
                 } else {
-                    unreachable!()
+                    unreachable!("We already know the operation is a query")
```

Comment:
we can do this in a different PR, but I feel like all the nested `if let else` statements in each match arm could be brought up a level to the match it self:

```rust
let operation = match (frame.opcode, frame.request_body()?) {
            (Opcode::Query, RequestBody::Query(body)) => {}
...
}
```

Then put the unreachable in the fall through arm.

Reply:
Unfortunately the entire reason we have to do this is because cassandra-protocol splits the logic into separate request_body and response_body methods.
I tried to unify them before but got resistance from the maintainer.
Its something to look into some day but I considered it low priority.